plugins/ftdidmx/FtdiDmxThread.cpp

```diff
@@ -27,29 +27,58 @@
 #include <unistd.h>
 
 #include <string>
+#include <queue>
+#include <utility>
 
 #include "ola/Clock.h"
 #include "ola/Logging.h"
 #include "ola/StringUtils.h"
+#include "ola/Constants.h"
+
+#include "ola/rdm/RDMCommand.h"
+#include "ola/rdm/RDMControllerInterface.h"
+#include "ola/rdm/RDMCommandSerializer.h"
+#include "ola/rdm/RDMResponseCodes.h"
+#include "ola/rdm/DiscoveryAgent.h"
+
 #include "plugins/ftdidmx/FtdiWidget.h"
 #include "plugins/ftdidmx/FtdiDmxThread.h"
 
 namespace ola {
 namespace plugin {
 namespace ftdidmx {
 
-FtdiDmxThread::FtdiDmxThread(FtdiInterface *interface, unsigned int frequency)
-  : m_granularity(UNKNOWN),
+FtdiDmxThread::FtdiDmxThread(FtdiInterface *interface,
+                             unsigned int frequency,
+                             unsigned int serial)
+  : m_timer("FtdiDmxThread"),
+    m_granularity(UNKNOWN),
     m_interface(interface),
     m_term(false),
-    m_frequency(frequency) {
+    m_frequency(frequency),
+    m_transaction_number(0),
+    m_discovery_agent(this),
+    m_uid(OPEN_LIGHTING_ESTA_CODE, serial),
+    m_pending_request(nullptr),
+    m_rdm_callback(nullptr),
+    m_mute_complete(nullptr),
+    m_unmute_complete(nullptr),
+    m_branch_callback(nullptr) {
+  m_timer.setCaller("FtdiDmxThread " + m_interface->Description());
+
+  if (serial == 0) {
+    std::srand(std::time(nullptr));
```

Comment:
You should use our existing random class really, which is cross-platform etc:
https://github.com/OpenLightingProject/ola/blob/master/common/math/Random.cpp